### ISSUE TITLE
Quality Review Reputation Pallet

### ISSUE THREAD

**Issue by Janislav** [chainflip-io/chainflip-backend#2181]
### Description

To prepare for Ibiza and Sandstorm we decided to give our pallets a little in-house audit. This issue includes a list of points that we should either consider changing or at least talk about and re-think.

### Findings

- [x] Heartbeat extrinsic is only ensure_signed - no check if an account is really a validator. We also boost the reputation of that account regardless of if it is a validator or just any account. I could imagine that this is a potential problem where an account can collect reputation before he even is a validator. His reputation is not getting deleted on the epoch expiration, because was not a validator in previous epochs.
- [x] #2217
- [x] Reputation has no connection to doing something actually useful. Maybe a proof of being a functional validator would make sense (think we talked about this earlier)
- [x] Non-validator submitting heartbeats is not really covered in the test suite. Maybe it's worth looking at how this could affect the system?
- [x] In the `reporting_any_offence_should_penalise_reputation_points_and_suspend` test case we treat BOB and ALICE in the same way even BOB is not a validator. The test doesn't take that into account. I know that's not a problem because, on that level, we don't really care about the state of an account (Validator, Nothing, Backup, etc). But due to the fact that we also don't prevent a non-validator from earning reputation at all this could be a problem, I guess.

Long story short: We should add a guard like`T::ValidatorManager::ensure_validator_or_backup()?` to prevent all of these problems (I guess).

### Alternatives Considered

None

### Additional context

Nothing

### Relevant people

@dandanlen
